.github/workflows/ci.yml

```diff
@@ -309,6 +309,16 @@ jobs:
       - name: Check if running in container
         if: matrix.container != ''
         run: echo "GHA_CONTAINER=${{ matrix.container }}" >> $GITHUB_ENV
+      - name: If running in container, upgrade packages
+        if: matrix.container != ''
+        run: |
+            apt-get -o Acquire::Retries=3 update && DEBIAN_FRONTEND=noninteractive apt-get -y install tzdata && apt-get -o Acquire::Retries=3 install -y sudo software-properties-common wget curl apt-transport-https make apt-file sudo unzip libssl-dev build-essential autotools-dev autoconf automake g++ libc++-helpers python ruby cpio gcc-multilib g++-multilib pkgconf python3 ccache libpython-dev
+            sudo apt-add-repository ppa:git-core/ppa
+            sudo apt-get -o Acquire::Retries=3 update && apt-get -o Acquire::Retries=3 -y install git
+            python_version=$(python3 -c 'import sys; print("{0.major}.{0.minor}".format(sys.version_info))')
+            sudo wget https://bootstrap.pypa.io/pip/$python_version/get-pip.py
+            sudo python3 get-pip.py
+            sudo /usr/local/bin/pip install cmake
```

Comment:
I have not had a need in this for a long while, IIRC Python 2.7, 3.4+ comes with Pip, strange to see that nowadays.

Reply:
@Kojoley , "a long while" is exactly the clue.  This script allows ubuntu 14 and ubuntu 16 to continue functioning a little while longer.   Otherwise pip is broken.   
Also, even on newer OS's, pip is a separate package, available but not automatically installed on docker containers.

Reply:
It feels like the condition then should be `if: os == "ubuntu-16.04" or os == "ubuntu-14.04"`, but there is no 14.04 image and 16.04 will be removed soon, so there is no point in the change.